### ISSUE TITLE
Page Lookup is a lot slow  in Version 1.2.7 

### ISSUE THREAD

**Issue by neerajbadlani** [oracle/opengrok#2760]
I moved to version 1.27 from 0.12 . And I see page lookups are lot slower ( no measurement , feels like it ) , 

* Also I have a browser open in portrait mode , in that case page never fully loads and I cant scroll the page down ( the problem doesnt happen in landscape mode ) . 
* Also top shows tomcat java process running at 100% entire time .

**Comment by vladak** [comment]
What sort of page lookups ? Cross-reference/history/annotation/directory view ? Most of the views except directory/annotation are pre-generated so the web app merely loads them from disk. Full CPU consumption by the webapp is definitely not expected - I am assuming there is no other web app running in your Tomcat. To see what is the webapp up to, run `jstack` and attach the result please.

**Comment by neerajbadlani** [comment]
Let me rephrase my question since I got to use 1.2.7 a little more . 
I dont think Page Lookup is slower . Just that in portrait mode , page never fulls loads up ( xref , annotate , . I see broken lines too . ( This happens only in Portrait Mode , Landscape no issues ) . Unfortunately
I cant upload screenshots of code pages . However below there is a screenshot of how things 
go awry in portrait mode . 

Search Bars get distorted in Portrait Mode : 
![image](https://user-images.githubusercontent.com/3440085/56973932-b1597200-6b22-11e9-8981-8181f59e2d9f.png)

History Page Numbers get distored in Portrait Mode : 
![image](https://user-images.githubusercontent.com/3440085/56974178-393f7c00-6b23-11e9-82f6-b215343a6a18.png)


JStack Logs as Requested for CPUHog
https://file.io/KCVIWu ( Dump 1 ) 
https://file.io/LUA0Hk ( Dump 2 )

**Comment by vladak** [comment]
What exactly is "Portrait mode" and which browser do you use ? What makes you think the page is not fully loaded ? The "Developer console" in Firefox will help to diagnose such issues, esp. the Network tab which will tell if there were any problems loading the content.

**Comment by vladak** [comment]
I am getting HTTP 404 error for the file.io link.

**Comment by neerajbadlani** [comment]
HI Vladak , Since the link was ephemeral , I think it deletes after first view , anyways I tried firefox and chrome incognito mode , in that it works fine , so maybe some plugin is probably causing an issue . 

http://osxdaily.com/2010/08/14/mac-setups-macbook-pro-with-a-24-inch-widescreen-in-portrait-mode/

Gist for CPUhog
https://gist.github.com/neerajbadlani/4457afcfa4580375f3a81fe6588f6374

**Comment by neerajbadlani** [comment]
Update for the portrait mode issue in case sb is interested : Just cleaned up cached images and files in Chrome . Now everything is ok .
